core/src/main/java/org/springframework/security/access/expression/AbstractSecurityExpressionHandler.java

```diff
@@ -75,6 +78,23 @@ public final EvaluationContext createEvaluationContext(Authentication authentica
 		return ctx;
 	}
 
+	/**
+	 * Invokes the internal template methods to create {@code StandardEvaluationContext}
+	 * and {@code SecurityExpressionRoot} objects.
+	 * @param authentication the {@code Supplier} of the current authentication object
+	 * @param invocation the invocation (filter, method, channel)
+	 * @return the context object for use in evaluating the expression, populated with a
+	 * suitable root object.
+	 */
+	@Override
+	public final EvaluationContext createEvaluationContext(Supplier<Authentication> authentication, T invocation) {
```

Comment:
While I appreciate the value of following the prevailing pattern, I think it would be better for `DefaultWeb`/`DefaultHttp`/`DefaultMethod`/`DefaultMessageSecurityExpressionHandler` to implement this method directly instead of replicating the protected methods.

Reply:
Obviously, we can add the protected methods later if that becomes necessary.

Reply:
Done.